homeassistant/components/fan/xiaomi_miio.py

```diff
@@ -196,50 +361,140 @@ def device_state_attributes(self):
 
     @property
     def is_on(self):
-        """Return true if fan is on."""
+        """Return true if device is on."""
         return self._state
 
-    @asyncio.coroutine
-    def _try_command(self, mask_error, func, *args, **kwargs):
-        """Call an air purifier command handling error messages."""
+    @staticmethod
+    def _extract_value_from_attribute(state, attribute):
+        from enum import Enum
+
+        value = getattr(state, attribute)
+        if isinstance(value, Enum):
+            return value.value
+
+        return value
+
+    async def _try_command(self, mask_error, func, *args, **kwargs):
+        """Call a miio device command handling error messages."""
         from miio import DeviceException
         try:
-            result = yield from self.hass.async_add_job(
+            result = await self.hass.async_add_job(
                 partial(func, *args, **kwargs))
 
-            _LOGGER.debug("Response received from air purifier: %s", result)
+            _LOGGER.debug("Response received from miio device: %s", result)
 
             return result == SUCCESS
         except DeviceException as exc:
             _LOGGER.error(mask_error, exc)
             return False
 
-    @asyncio.coroutine
-    def async_turn_on(self: ToggleEntity, speed: str = None, **kwargs) -> None:
-        """Turn the fan on."""
+    async def async_turn_on(self: ToggleEntity, speed: str = None,
```

Comment:
Should I remove the return type (ToggleEntity) here? `self._device` isn't part of the ToggleEntity. pycharm is complaining about.

Reply:
Yes, there is no need to add typing hints for `self`.